src/spikeinterface/curation/tests/test_auto_merge.py

```diff
@@ -81,60 +82,82 @@ def test_compute_merge_unit_groups(sorting_analyzer_for_curation, preset):
+#     unit_ind2 = sorting_with_split.id_to_index(unit_id2)
 
-    #     ax.set_title(f'{unit_id1} {unit_id2}')
-    #     ax = axs[1]
-    #     ax.plot(bins2, correlograms_smoothed[unit_ind1, unit_ind2, :], color='g')
+#     bins2 = bins[:-1] + np.mean(np.diff(bins))
+#     fig, axs = plt.subplots(ncols=3)
+#     ax = axs[0]
+#     ax.plot(bins2, correlograms[unit_ind1, unit_ind1, :], color='b')
+#     ax.plot(bins2, correlograms[unit_ind2, unit_ind2, :], color='r')
+#     ax.plot(bins2, correlograms_smoothed[unit_ind1, unit_ind1, :], color='b')
+#     ax.plot(bins2, correlograms_smoothed[unit_ind2, unit_ind2, :], color='r')
 
-    #     auto_corr1 = normalize_correlogram(correlograms_smoothed[unit_ind1, unit_ind1, :])
-    #     auto_corr2 = normalize_correlogram(correlograms_smoothed[unit_ind2, unit_ind2, :])
-    #     cross_corr = normalize_correlogram(correlograms_smoothed[unit_ind1, unit_ind2, :])
+#     ax.set_title(f'{unit_id1} {unit_id2}')
+#     ax = axs[1]
+#     ax.plot(bins2, correlograms_smoothed[unit_ind1, unit_ind2, :], color='g')
 
-    #     ax = axs[2]
-    #     ax.plot(bins2, auto_corr1, color='b')
-    #     ax.plot(bins2, auto_corr2, color='r')
-    #     ax.plot(bins2, cross_corr, color='g')
+#     auto_corr1 = normalize_correlogram(correlograms_smoothed[unit_ind1, unit_ind1, :])
+#     auto_corr2 = normalize_correlogram(correlograms_smoothed[unit_ind2, unit_ind2, :])
+#     cross_corr = normalize_correlogram(correlograms_smoothed[unit_ind1, unit_ind2, :])
 
-    #     ax.axvline(bins2[m - win_sizes[unit_ind1]], color='b')
-    #     ax.axvline(bins2[m + win_sizes[unit_ind1]], color='b')
-    #     ax.axvline(bins2[m - win_sizes[unit_ind2]], color='r')
-    #     ax.axvline(bins2[m + win_sizes[unit_ind2]], color='r')
+#     ax = axs[2]
+#     ax.plot(bins2, auto_corr1, color='b')
+#     ax.plot(bins2, auto_corr2, color='r')
+#     ax.plot(bins2, cross_corr, color='g')
 
-    #     ax.set_title(f'corr diff {correlogram_diff[unit_ind1, unit_ind2]} - temp diff {templates_diff[unit_ind1, unit_ind2]}')
-    #     plt.show()
+#     ax.axvline(bins2[m - win_sizes[unit_ind1]], color='b')
+#     ax.axvline(bins2[m + win_sizes[unit_ind1]], color='b')
+#     ax.axvline(bins2[m - win_sizes[unit_ind2]], color='r')
+#     ax.axvline(bins2[m + win_sizes[unit_ind2]], color='r')
+
+#     ax.set_title(f'corr diff {correlogram_diff[unit_ind1, unit_ind2]} - temp diff {templates_diff[unit_ind1, unit_ind2]}')
+#     plt.show()
+
+
+def test_auto_merge_units(sorting_analyzer_for_curation):
+    recording = sorting_analyzer_for_curation.recording
+    job_kwargs = dict(n_jobs=-1)
+    new_sorting, _ = split_sorting_by_times(sorting_analyzer_for_curation)
+    new_sorting_analyzer = create_sorting_analyzer(new_sorting, recording, format="memory")
+    merged_analyzer = auto_merge_units(new_sorting_analyzer, presets="x_contaminations", **job_kwargs)
+    assert len(merged_analyzer.unit_ids) < len(new_sorting_analyzer.unit_ids)
+
```

Comment:
```suggestion
    merged_analyzer = auto_merge_units(new_sorting_analyzer, presets="x_contaminations", **job_kwargs)
    assert len(merged_analyzer.unit_ids) < len(new_sorting_analyzer.unit_ids)
    
    # test for steps integration
    step_merged_analyzer = auto_merge_units(new_sorting_analyzer, steps = ["num_spikes", "removed_contaminated", "unit_locations", "template_similarity", "quality_scores"], step_params = dict(num_spikes=150), **job_kwargs)
    assert len(step_merged_analyzer.unit_ids) < len(new_sorting_analyzer.unit_ids)

```

what about something like this? Might have a typo so we may need to fix since it is hard to code in this code review window? My goal here is to :
1) test for steps instead of presets to make sure the steps flow is sound
2) test changing a preset in a way I imagine a user might want to do

Reply:
This has been done, I've added the test and I consolidated the functions such that it could be used the way you thought